c++/src/Reader.hh

```diff
@@ -260,6 +268,13 @@ namespace orc {
     // footer
     proto::Footer* footer_;
     uint64_t numberOfStripes_;
+
+    // cached io ranges. only valid when preBuffer is invoked.
+    std::shared_ptr<ReadRangeCache> readCache_;
+
+    // mutex to protect readCache_ from concurrent access
+    std::mutex readCacheMutex_;
```

Comment:
I think this is the proper place of mutex. Because the mutex could not only protect the interfaces of `ReadRangeCache`, but also the creation of `ReadRangeCache`

Reply:
Make sense. I have noticed this since last review but forgot to reply to my own comment.